lib/Dialect/OM/Evaluator/Evaluator.cpp

```diff
@@ -64,43 +82,43 @@ circt::om::Evaluator::instantiate(StringAttr className,
     if (actualParamType != formalParamType) {
       auto error = cls.emitError("actual parameter for ")
                    << formalParamName << " has invalid type";
-      error.attachNote() << "actual parameter: " << actualParam;
+      error.attachNote() << "actual parameter: " << *actualParam;
       error.attachNote() << "format parameter type: " << formalParamType;
       return error;
     }
   }
 
   // Instantiate the fields.
-  ObjectFields fields;
+  evaluator::ObjectFields fields;
   for (auto field : cls.getOps<ClassFieldOp>()) {
     StringAttr name = field.getSymNameAttr();
     Value value = field.getValue();
 
-    FailureOr<ObjectValue> result = evaluateValue(value, actualParams);
+    FailureOr<evaluator::EvaluatorValuePtr> result =
+        evaluateValue(value, actualParams);
     if (failed(result))
       return failure();
 
     fields[name] = result.value();
   }
 
   // Allocate the Object. Further refinement is expected.
-  auto *object = new Object(cls, fields);
+  auto *object = new evaluator::ObjectValue(cls, fields);
```

Comment:
I tried llvm::BumpAllocator but bumpAllocator frees entire memories in the destructor so it seems not usable with smart pointers.

Reply:
Interesting... we can definitely separate the allocation improvements from the current change to generalize and improve the types at runtime.